src/main/java/duke/Execution.java

```diff
@@ -0,0 +1,53 @@
+package duke;
+
+import duke.command.*;
+import duke.exception.CommandException;
+
+public class Execution {
+    protected String fullCommand;
+    protected String command;
+    protected boolean isExit;
+
+    public Execution(String fullCommand) {
+        this.fullCommand = fullCommand;
+        this.command = Parser.command(fullCommand);
+        this.isExit = false;
+    }
+
+    private static void CheckCommand(String command) throws CommandException {
+        if ( !(command.equals("todo" ) || command.equals( "deadline") || command.equals( "event")
+                || command.equals( "bye") || command.equals( "list") || command.equals( "done")) ) {
+            throw new CommandException();
+        }
+    }
+
+    public void execute(TaskList taskList) {
+        try {
+            CheckCommand(command);
+        } catch (CommandException e) {
+            System.out.println("OOPS!!! Pls key in the valid command");
+        }
+
+        switch (command) {
```

Comment:
Move switch cases into the try block instead.

Reply:
Good advice. Will improve code accordingly.